install fails - cmake version incompat in Poly-EP

`brew install cvc5/homebrew-cvc5/cvc5` fails on my M3 MBP. Looks like brew is running a new-enough version of cmake that it can no longer build Poly-EP. Full log:

```
2025-04-28 20:04:07 +0000

make
install

[  0%] Building CXX object src/context/CMakeFiles/cvc5context.dir/context.cpp.o
[  0%] Generating rewrites.{h,cpp}
[  1%] Generating theory_traits.h
[  1%] Creating directories for 'SymFPU-EP'
[  1%] Generating smt2_tokens.h
[  1%] Generating options/options.stamp
[  1%] Generating node_manager.h
[  1%] Creating directories for 'Poly-EP'
[  1%] Generating rewriter_tables.h
[  2%] Building CXX object src/context/CMakeFiles/cvc5context.dir/context_mm.cpp.o
[  2%] Generating kind.h
[  2%] Generating metakind.h
[  2%] Generating type_enumerator.cpp
[  2%] Creating directories for 'CaDiCaL-EP'
[  2%] Generating Trace_tags.h
[  2%] Performing download step (download, verify and extract) for 'SymFPU-EP'
[  2%] Performing download step (download, verify and extract) for 'Poly-EP'
[  2%] Built target gen-tokens
[  2%] Performing download step (download, verify and extract) for 'CaDiCaL-EP'
[  2%] Generating type_checker.cpp
-- Found Git: /opt/homebrew/Library/Homebrew/shims/mac/super/git (found version "2.39.5 (Apple Git-154)")
[  2%] Built target gen-options
[  2%] Generating type_properties.h
[  2%] Built target gen-versioninfo
[  2%] Built target gen-tags
[  3%] Building CXX object src/base/CMakeFiles/cvc5base.dir/configuration.cpp.o
[  3%] Building CXX object src/base/CMakeFiles/cvc5base.dir/check.cpp.o
[  3%] Building CXX object src/base/CMakeFiles/cvc5base.dir/exception.cpp.o
[  3%] Building CXX object src/base/CMakeFiles/cvc5base.dir/listener.cpp.o
[  3%] Built target cvc5context
[  3%] Building CXX object src/base/CMakeFiles/cvc5base.dir/output.cpp.o
[  3%] Building CXX object src/base/CMakeFiles/cvc5base.dir/versioninfo.cpp.o
[  3%] Built target gen-theory
[  3%] Built target cvc5base
[  3%] Generating kind.cpp
-- SymFPU-EP download command succeeded.  See also /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/SymFPU-EP-stamp/SymFPU-EP-download-*.log
[  3%] Generating type_properties.cpp
[  3%] No update step for 'SymFPU-EP'
[  3%] No patch step for 'SymFPU-EP'
[  3%] No configure step for 'SymFPU-EP'
[  3%] No build step for 'SymFPU-EP'
[  4%] Performing install step for 'SymFPU-EP'
-- SymFPU-EP install command succeeded.  See also /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/SymFPU-EP-stamp/SymFPU-EP-install-*.log
[  4%] Completed 'SymFPU-EP'
[  4%] Built target SymFPU-EP
[  4%] Built target gen-rewrites
-- CaDiCaL-EP download command succeeded.  See also /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/CaDiCaL-EP-stamp/CaDiCaL-EP-download-*.log
[  4%] No update step for 'CaDiCaL-EP'
[  4%] No patch step for 'CaDiCaL-EP'
[  4%] Performing configure step for 'CaDiCaL-EP'
-- CaDiCaL-EP configure command succeeded.  See also /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/CaDiCaL-EP-stamp/CaDiCaL-EP-configure-*.log
[  4%] Performing build step for 'CaDiCaL-EP'
-- Poly-EP download command succeeded.  See also /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/Poly-EP-stamp/Poly-EP-download-*.log
[  4%] No update step for 'Poly-EP'
[  4%] Performing patch step for 'Poly-EP'
-- Poly-EP patch command succeeded.  See also /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/Poly-EP-stamp/Poly-EP-patch-*.log
[  5%] Performing configure step for 'Poly-EP'
CMake Error at /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/Poly-EP-stamp/Poly-EP-configure-Production.cmake:37 (message):
  Command failed: 1

   '/opt/homebrew/opt/cmake/bin/cmake' '-DCMAKE_BUILD_TYPE=Release' '-DCMAKE_INSTALL_PREFIX=/private/tmp/cvc5-20250428-15816-5kq1m2/build/deps' '-DCMAKE_TOOLCHAIN_FILE=' '-DLIBPOLY_BUILD_PYTHON_API=OFF' '-DLIBPOLY_BUILD_STATIC=ON' '-DLIBPOLY_BUILD_STATIC_PIC=ON' '-DGMP_INCLUDE_DIR=/opt/homebrew/include' '-DGMP_LIBRARY=/opt/homebrew/lib/libgmp.a' '-DCMAKE_SKIP_INSTALL_ALL_DEPENDENCY=TRUE' '-GUnix Makefiles' '-S' '/private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/Poly-EP' '-B' '/private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/Poly-EP-build'

  See also

    /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/Poly-EP-stamp/Poly-EP-configure.log


-- Log output is:
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!

CMake Error at /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/Poly-EP-stamp/Poly-EP-configure-Production.cmake:47 (message):
  Stopping after outputting logs.


make[2]: *** [deps/src/Poly-EP-stamp/Poly-EP-configure] Error 1
make[1]: *** [CMakeFiles/Poly-EP.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  5%] Generating metakind.cpp
[  5%] Generating node_manager.cpp
[  5%] Built target gen-expr
-- CaDiCaL-EP build command succeeded.  See also /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/CaDiCaL-EP-stamp/CaDiCaL-EP-build-*.log
[  5%] Performing install step for 'CaDiCaL-EP'
-- CaDiCaL-EP install command succeeded.  See also /private/tmp/cvc5-20250428-15816-5kq1m2/build/deps/src/CaDiCaL-EP-stamp/CaDiCaL-EP-install-*.log
[  5%] Completed 'CaDiCaL-EP'
[  5%] Built target CaDiCaL-EP
make: *** [all] Error 2

HOMEBREW_VERSION: 4.4.29
ORIGIN: https://github.com/Homebrew/brew
HEAD: 12a3d4a6f1eedf483855716b989d828443438f79
Last commit: 6 days ago
Branch: stable
Core tap JSON: 28 Apr 20:02 UTC
Core cask tap JSON: 28 Apr 20:02 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_NO_BOOTSNAP: set
Homebrew Ruby: 3.3.8 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.8/bin/ruby
CPU: 16-core 64-bit arm_palma
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.3.2-arm64
CLT: 16.2.0.0.1.1733547573
Xcode: N/A
Rosetta 2: false

HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
MAKEFLAGS: -j16
CMAKE_PREFIX_PATH: /opt/homebrew
CMAKE_INCLUDE_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
CMAKE_FRAMEWORK_PATH: /opt/homebrew/opt/python@3.13/Frameworks
PKG_CONFIG_PATH: /opt/homebrew/opt/python@3.13/lib/pkgconfig:/opt/homebrew/opt/gmp/lib/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/15
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk
ACLOCAL_PATH: /opt/homebrew/share/aclocal
PATH: /opt/homebrew/Library/Homebrew/shims/mac/super:/opt/homebrew/opt/cmake/bin:/opt/homebrew/opt/python@3.13/bin:/opt/homebrew/opt/python@3.13/libexec/bin:/usr/bin:/bin:/usr/sbin:/sbin

```

--- a/Casks/cvc5.rb
+++ b/Casks/cvc5.rb
@@ -2,11 +2,23 @@ cask "cvc5" do
   arch arm: "arm64", intel: "x86_64"
 
   version "1.2.1"
-  sha256 arm:   "18e0bd283d44f720f72bf80175169ef63e985628b7ba1502aaf812f57f981461",
-         intel: "bdde9557bbb9b812af270c3f418836c285957b3ed81385b87428d2d595ffbf47"
 
-  url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-macOS-#{arch}-static.zip",
-      verified: "github.com/cvc5/cvc5/releases/download/"
+  on_macos do
+    url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-macOS-#{arch}-static.zip",
+        verified: "github.com/cvc5/cvc5/releases/download/"
+    sha256 arm:   "18e0bd283d44f720f72bf80175169ef63e985628b7ba1502aaf812f57f981461",
+           intel: "bdde9557bbb9b812af270c3f418836c285957b3ed81385b87428d2d595ffbf47"
+    binary "cvc5-macOS-#{arch}-static/bin/cvc5"
+  end
+
+  on_linux do
+    url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-Linux-#{arch}-static.zip",
+        verified: "github.com/cvc5/cvc5/releases/download/"
+    sha256 arm:   "de27e95e00d16fbd7697ab7eac3046d0a11ca8753e2f50c5ed52604e75691291",
+           intel: "6d44abc233980a14d72cc5809287d27c3335b1d6ee863381d0b5ffcbd0d8de56"
+    binary "cvc5-Linux-#{arch}-static/bin/cvc5"
+  end
+
   name "cvc5"
   desc "Automatic theorem prover for Satisfiability Modulo Theories (SMT) problems"
   homepage "https://cvc5.github.io/"
@@ -17,8 +29,6 @@ cask "cvc5" do
     strategy :github_latest
     regex(/^cvc5-(\d+(?:\.\d+)+)$/i)
   end
-
-  binary "cvc5-macOS-#{arch}-static/bin/cvc5"
 
   caveats do
     license "https://github.com/cvc5/cvc5/blob/main/COPYING"

--- a/Casks/cvc5.rb
+++ b/Casks/cvc5.rb
@@ -1,0 +1,25 @@
+cask "cvc5" do
+  arch arm: "arm64", intel: "x86_64"
+  
+  version "1.2.1"
+  sha256 arm:   "18e0bd283d44f720f72bf80175169ef63e985628b7ba1502aaf812f57f981461",
+         intel: "bdde9557bbb9b812af270c3f418836c285957b3ed81385b87428d2d595ffbf47"
+
+  url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-macOS-#{arch}-static.zip"
+  name "cvc5"
+  desc "Automatic theorem prover for Satisfiability Modulo Theories (SMT) problems"
+  homepage "https://cvc5.github.io/"
+  
+  # Use GitHub releases to check for new versions
+  livecheck do
+    url :url
+    strategy :github_latest
+    regex(/^cvc5-(\d+(?:\.\d+)+)$/i)
+  end
+  
+  binary "cvc5-macOS-#{arch}-static/bin/cvc5"
+  
+  caveats do
+    license "https://github.com/cvc5/cvc5/blob/main/COPYING"
+  end
+end

--- a/Casks/cvc5.rb
+++ b/Casks/cvc5.rb
@@ -5,7 +5,7 @@ cask "cvc5" do
   sha256 arm:   "18e0bd283d44f720f72bf80175169ef63e985628b7ba1502aaf812f57f981461",
          intel: "bdde9557bbb9b812af270c3f418836c285957b3ed81385b87428d2d595ffbf47"
 
-  url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-macOS-#{arch}-static.zip"
+  url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-macOS-#{arch}-static.zip",
       verified: "github.com/cvc5/cvc5/releases/download/"
   name "cvc5"
   desc "Automatic theorem prover for Satisfiability Modulo Theories (SMT) problems"

--- a/Casks/cvc5.rb
+++ b/Casks/cvc5.rb
@@ -1,9 +1,8 @@
 cask "cvc5" do
-  arch arm: "arm64", intel: "x86_64"
-
   version "1.2.1"
 
   on_macos do
+    arch arm: "arm64", intel: "x86_64"
     url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-macOS-#{arch}-static.zip",
         verified: "github.com/cvc5/cvc5/releases/download/"
     sha256 arm:   "18e0bd283d44f720f72bf80175169ef63e985628b7ba1502aaf812f57f981461",
@@ -12,11 +11,10 @@ cask "cvc5" do
   end
 
   on_linux do
-    url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-Linux-#{arch}-static.zip",
+    url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-Linux-x86_64-static.zip",
         verified: "github.com/cvc5/cvc5/releases/download/"
-    sha256 arm:   "de27e95e00d16fbd7697ab7eac3046d0a11ca8753e2f50c5ed52604e75691291",
-           intel: "6d44abc233980a14d72cc5809287d27c3335b1d6ee863381d0b5ffcbd0d8de56"
-    binary "cvc5-Linux-#{arch}-static/bin/cvc5"
+    sha256 "6d44abc233980a14d72cc5809287d27c3335b1d6ee863381d0b5ffcbd0d8de56"
+    binary "cvc5-Linux-x86_64-static/bin/cvc5"
   end
 
   name "cvc5"

--- a/Casks/cvc5.rb
+++ b/Casks/cvc5.rb
@@ -1,24 +1,25 @@
 cask "cvc5" do
   arch arm: "arm64", intel: "x86_64"
-  
+
   version "1.2.1"
   sha256 arm:   "18e0bd283d44f720f72bf80175169ef63e985628b7ba1502aaf812f57f981461",
          intel: "bdde9557bbb9b812af270c3f418836c285957b3ed81385b87428d2d595ffbf47"
 
   url "https://github.com/cvc5/cvc5/releases/download/cvc5-#{version}/cvc5-macOS-#{arch}-static.zip"
+      verified: "github.com/cvc5/cvc5/releases/download/"
   name "cvc5"
   desc "Automatic theorem prover for Satisfiability Modulo Theories (SMT) problems"
   homepage "https://cvc5.github.io/"
-  
+
   # Use GitHub releases to check for new versions
   livecheck do
     url :url
     strategy :github_latest
     regex(/^cvc5-(\d+(?:\.\d+)+)$/i)
   end
-  
+
   binary "cvc5-macOS-#{arch}-static/bin/cvc5"
-  
+
   caveats do
     license "https://github.com/cvc5/cvc5/blob/main/COPYING"
   end

--- a/README.md
+++ b/README.md
@@ -1,10 +1,16 @@
 # Cvc5
 
-## How do I install these formulae?
+## How do I install cvc5 using a formula?
 
-`brew install cvc5/homebrew-cvc5/cvc5`
+`brew install cvc5/cvc5/cvc5`
 
-Or `brew tap cvc5/homebrew-cvc5` and then `brew install cvc5`.
+Or `brew tap cvc5/cvc5` and then `brew install cvc5`.
+
+## How do I install cvc5 as a cask?
+
+`brew install --cask cvc5/cvc5/cvc5`
+
+Or `brew tap cvc5/cvc5` and then `brew install --cask cvc5`.
 
 ## Documentation
 

